src/Features/CSharp/Portable/CodeFixes/AddExplicitCast/AddExplicitCastCodeFixProvider.cs

```diff
@@ -0,0 +1,399 @@
+        /// False, if the target node has no conversion type.
+        /// </returns>
+        private static bool TryGetTargetTypeInfo(SemanticModel semanticModel, string diagnosticId, SyntaxNode targetNode, CancellationToken cancellationToken,
+            [NotNullWhen(true)]  out ITypeSymbol? targetNodeType, out ImmutableArray<ITypeSymbol> potentialConversionTypes)
+        {
+            potentialConversionTypes = ImmutableArray<ITypeSymbol>.Empty;
+
+            var targetNodeInfo = semanticModel.GetTypeInfo(targetNode, cancellationToken);
+            targetNodeType = targetNodeInfo.Type;
+
+            if (targetNodeType == null)
+            {
+                return false;
+            }
+
+            // The error happens either on an assignement operation or on an invocation expression.
+            // If the error happens on assignment operation, "ConvertedType" is different from the current "Type"
+            var mutablePotentialConversionTypes = ArrayBuilder<ITypeSymbol>.GetInstance();
+            if (diagnosticId == "CS0266" && targetNodeInfo.ConvertedType != null && !targetNodeType.Equals(targetNodeInfo.ConvertedType))
+            {
+                mutablePotentialConversionTypes.Add(targetNodeInfo.ConvertedType);
+            }
+            else if (diagnosticId == "CS1503" && targetNode.GetAncestorsOrThis<ArgumentSyntax>().FirstOrDefault() is ArgumentSyntax targetArgument
+                && targetArgument.Parent is ArgumentListSyntax argumentList
+                && argumentList.Parent is SyntaxNode invocationNode) // invocation node could be Invocation Expression, Object Creation, Base Constructor...
+            {
+                // Implicit downcast appears on the argument of invocation node, get all candidate functions and extract potential conversion types 
+                var symbolInfo = semanticModel.GetSymbolInfo(invocationNode, cancellationToken);
+                var candidateSymbols = symbolInfo.CandidateSymbols;
+
+                foreach (var candidateSymbol in candidateSymbols)
+                {
+                    if (!(candidateSymbol is IMethodSymbol methodSymbol))
+                    {
+                        continue;
+                    }
+
+                    if (IsArgumentListAndParameterListPerfectMatch(semanticModel, argumentList.Arguments, methodSymbol.Parameters, targetArgument, cancellationToken, out var paramIndex))
+                    {
+                        var correspondingParameter = methodSymbol.Parameters[paramIndex];
+                        var argumentConversionType = correspondingParameter.Type;
+
+                        if (correspondingParameter.IsParams && correspondingParameter.Type is IArrayTypeSymbol arrayType && !(targetNodeType is IArrayTypeSymbol))
+                        {
+                            // target argument is matched to the parameter with keyword params
+                            argumentConversionType = arrayType.ElementType;
```

Comment:
nice.  do you have tests for this?

Reply:
Tests for keyword `params`? Yes, I have a lot :)

Reply:
awsome :)